multiapi-engine/src/main/java/com/sngular/api/generator/plugin/asyncapi/template/TemplateFactory.java

```diff
@@ -117,8 +119,154 @@ public final void fillTemplateModelClassException(final Path filePathToSave, fin
     pathToExceptionPackage.toFile().mkdirs();
     root.put("packageModel", modelPackage);
     final String pathToSaveMainClass = pathToExceptionPackage.resolve("ModelClassException.java").toString();
-    writeTemplateToFile(com.sngular.api.generator.plugin.openapi.template.TemplateIndexConstants.TEMPLATE_MODEL_EXCEPTION, root, pathToSaveMainClass);
+    writeTemplateToFile(TemplateIndexConstants.TEMPLATE_MODEL_EXCEPTION, root, pathToSaveMainClass);
+  }
+
+  public final void fillTemplateNotNull(final Path filePathToSave, final String modelPackage) throws IOException, TemplateException {
+    final Path pathToValidatorPackage = filePathToSave.resolve("customvalidator");
+    pathToValidatorPackage.toFile().mkdirs();
+    root.put("packageModel", modelPackage);
+    final String pathToSaveMainClass = pathToValidatorPackage.resolve("NotNull.java").toString();
+    writeTemplateToFile(TemplateIndexConstants.TEMPLATE_NOT_NULL_ANNOTATION, root, pathToSaveMainClass);
+  }
```

Comment:
All of these methods seem very similar, can a common method be extracted and parametrized?

Reply:
Done